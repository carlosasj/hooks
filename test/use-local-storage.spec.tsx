import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';

import useLocalStorage from '../src/use-local-storage';

describe('useLocalStorage', () => {
  const Component = () => {
    const [storedValue, setValue] = useLocalStorage('data', { a: 1 });

    const handleClick = React.useCallback(() => {
      setValue({ b: 2 });
    }, [setValue]);

    return (
      <div>
        <h1>{JSON.stringify(storedValue, null, 2)}</h1>
        <button type="button" onClick={handleClick}>
          Update
        </button>
      </div>
    );
  };

  const { container } = render(<Component />);

  it('should render the data and re-render when updated', () => {
    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(container.querySelector('button')!);

    expect(container.firstChild).toMatchSnapshot();
  });
});
