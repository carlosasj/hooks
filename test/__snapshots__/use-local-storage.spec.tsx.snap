// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useLocalStorage should render the data and re-render when updated 1`] = `
<div>
  <h1>
    {
  "a": 1
}
  </h1>
  <button
    type="button"
  >
    Update
  </button>
</div>
`;

exports[`useLocalStorage should render the data and re-render when updated 2`] = `
<div>
  <h1>
    {
  "b": 2
}
  </h1>
  <button
    type="button"
  >
    Update
  </button>
</div>
`;
